server/repo/repo-core/src/main/java/org/eclipse/vorto/repository/service/IModelRepository.java

```diff
@@ -67,6 +68,15 @@
 	UploadModelResult upload(byte[] content, String fileName);
 	
 	/**
+	 * Uploads multiple model files in zip and validates it.
+	 * If all the models are valid then the models are checked in using array of {@code} ModelHandle 
+	 * {@link IModelRepository#checkinMultiple(ModelHandle)}}
+	 * @param fileName
+	 * @return result about information of the uploaded content and the upload handle. 
+	 */
+	List<UploadModelResult> uploadMultipleModels(String fileName);
```

Comment:
Yes. The existing single file upload validations criteria are different which is not suitable for multiple model upload and also to avoid multiple conditional checks for single and multiple upload files.